c++/h5/CMakeLists.txt

```diff
@@ -27,13 +27,16 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DESTINATION include FILES_MATCHING
 message(STATUS "-------- HDF5 detection -------------")
 
 find_package(HDF5 REQUIRED C HL)
+if(NOT HDF5_VERSION)
+  set(HDF5_SUBVERSION_ISSUE TRUE)
+endif()
 
 # Link against interface target and export
 target_link_libraries(h5_c PRIVATE hdf5::hdf5 hdf5::hdf5_hl)
-if(HDF5_VERSION VERSION_GREATER 1.10)
+if(HDF5_VERSION VERSION_GREATER 1.10 OR HDF5_SUBVERSION_ISSUE)
   target_compile_definitions(h5_c PRIVATE H5_USE_110_API)
 endif()
-if(HDF5_VERSION VERSION_GREATER_EQUAL 1.13)
+if(HDF5_VERSION VERSION_GREATER_EQUAL 1.13 OR HDF5_SUBVERSION_ISSUE)
   target_compile_definitions(h5_c PRIVATE H5_VER_GE_113)
 endif()
```

Comment:
I wonder why we need the version detection at all. The names of the macros do not suggest that these macros would have any effect on older versions at all, so I think the most pragmatic solution would be to enable these unconditionally.

Reply:
Another option would be to negate the logic, i.e.
```cmake
if(NOT HDF5_VERSION VERSION_LESS 1.10)
  target_compile_definitions(h5_c PRIVATE H5_USE_110_API)
endif()
if(NOT HDF5_VERSION VERSION_LESS 1.13)
  target_compile_definitions(h5_c PRIVATE H5_VER_GE_113)
endif()
```
under the assumption that the HDF5 version detection is not broken for these older versions.